modules/programs/git.nix

```diff
@@ -188,9 +194,13 @@ in
       (mkIf (cfg.includes != []) {
         xdg.configFile."git/config".text = mkAfter
           (concatMapStringsSep "\n"
-            (i: with i; ''
+            (i: with i;
+            let
+              path' = if (contents != {}) then (pkgs.writeText "contents" (generators.toINI {} contents)) else path;
```

Comment:
Instead of assembling `path'` here you can use the same trick as in [`file-type.nix`](https://github.com/rycee/home-manager/blob/0590c2a4f64871d00487639220d04955194bc172/modules/lib/file-type.nix#L93-L98) to default the `path` of the submodule to the INI content.

Reply:
thanks for pointing this out! this was new to me :)